server/leader.go

```diff
@@ -30,22 +29,36 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-var (
-	errNoLeader = errors.New("no leader")
-)
-
 // IsLeader returns whether server is leader or not.
 func (s *Server) IsLeader() bool {
-	return atomic.LoadInt64(&s.isLeader) == 1
+	// If server is not started. Both leaderID and ID could be 0.
+	return !s.isClosed() && s.GetLeaderID() == s.ID()
+}
+
+// GetLeaderID returns current leader's member ID.
+func (s *Server) GetLeaderID() uint64 {
+	return s.GetLeader().GetMemberId()
 }
 
-func (s *Server) enableLeader(b bool) {
-	value := int64(0)
-	if b {
-		value = 1
+// GetLeader returns current leader of pd cluster.
+func (s *Server) GetLeader() *pdpb.Member {
+	leader := s.leader.Load()
+	if leader == nil {
+		return nil
+	}
+	member := leader.(*pdpb.Member)
+	if member.GetMemberId() == 0 {
+		return nil
 	}
+	return member
+}
+
+func (s *Server) enableLeader() {
+	s.leader.Store(s.member)
+}
 
-	atomic.StoreInt64(&s.isLeader, value)
+func (s *Server) disableLeader() {
+	s.leader.Store(&pdpb.Member{})
```

Comment:
why not setting nil directly?

Reply:
It will panic if you store nil. See https://godoc.org/sync/atomic#Value.Store